src/Eventso.Subscription.Kafka/DeadLetter/RetryingEventHandler.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Eventso.Subscription.Kafka.DeadLetter.Store;
+using Eventso.Subscription.Observing.DeadLetter;
+
+namespace Eventso.Subscription.Kafka.DeadLetter
+{
+    public sealed class RetryingEventHandler : IEventHandler<Event>
+    {
+        private readonly IEventHandler<Event> _inner;
+        private readonly IDeadLetterQueueScopeFactory _deadLetterQueueScopeFactory;
+        private readonly IPoisonEventStore _poisonEventStore;
+
+        public RetryingEventHandler(
+            IEventHandler<Event> inner,
+            IDeadLetterQueueScopeFactory deadLetterQueueScopeFactory,
+            IPoisonEventStore poisonEventStore)
+        {
+            _inner = inner;
+            _deadLetterQueueScopeFactory = deadLetterQueueScopeFactory;
+            _poisonEventStore = poisonEventStore;
+        }
+
+        public async Task Handle(Event @event, CancellationToken cancellationToken)
+        {
+            using var dlqScope = _deadLetterQueueScopeFactory.Create(@event);
+
+            OccuredFailure? occuredFailure = null; 
+            try
+            {
+                await _inner.Handle(@event, cancellationToken);
+            }
+            catch (Exception exception)
+            {
+                occuredFailure = new OccuredFailure(@event.GetTopicPartitionOffset(), exception.ToString());
+            }
```

Comment:
`OccuredFailure? occuredFailure = null;` does not look cool. just extract a method with this try-catch